channeldb/invoices.go

```diff
@@ -1148,38 +1145,42 @@ func fetchInvoice(invoiceNum []byte, invoices kvdb.RBucket,
 		return invpkg.Invoice{}, err
 	}
 
-	// If this is an AMP invoice, then we'll also attempt to read out the
-	// set of HTLCs that were paid to prior set IDs. However, we'll only do
-	// this is the invoice didn't already have HTLCs stored in-line.
-	invoiceIsAMP := invoice.Terms.Features.HasFeature(
-		lnwire.AMPOptional,
-	)
-	switch {
-	case !invoiceIsAMP:
-		return invoice, nil
+	// If this is an AMP invoice we'll also attempt to read out the set of
```

Comment:
I think this would read _a little_ cleaner if we first have: 
```
if !invoice.IsAMP(){
    return invoice, nil
}
```

Then go on to fetch the AMP HTLCs if we need them?

Reply:
I also like it more like 
```
if !invoice.IsAMP(){
    return invoice, nil
}
``` 

I changed it so in the future we can easily add new types of invoice:

```
i := getSharedInvoiceData()
if i.IsAmp() {
    getAmpSpecificData()
}
if i.IsBolt12 {
    getBolt12SpecificData()
}
...


return i, nil
```

Maybe is an example of YAGNI?

Reply:
Can probably just start with exiting early on `!isAmp` so that we can cut one of the condition checks? Small change to refactor further if more invoice formats are added :)